### PR TITLE
DEV-2023: Fixed Bulk Download API bug for sub-awards

### PR DESCRIPTION
**Pull request by tony-sappe**
**Description:**
Added an explicit reference to the subaward materialized view to resolve a bug when trying to use the endpoint to generate downloads.

**Technical details:**
A line to add a SQL fragment caused ambiguity in the column reference.

**Requirements for PR merge:**

1. [x] Unit & integration tests updated (N/A)
2. [x] API documentation updated (N/A)
3. [x] Necessary PR reviewers:
    - [x] Backend
4. [x] Matview impact assessment completed (N/A)
5. [x] Frontend impact assessment completed (N/A)
6. [x] Data validation completed (N/A)
7. [x] Appropriate Operations ticket(s) created (N/A)
8. [x] Jira Ticket [DEV-2023](https://federal-spending-transparency.atlassian.net/browse/DEV-2023):
    - [x] Link to this Pull-Request
    - [x] Performance evaluation of affected Download
    - [x] Before / After data comparison (N/A)

**Area for explaining above N/A when needed:**
```
No API behavior changes except correcting an error causing downloads to fail.
No impact to operations or frontend
```

### CODE REVIEW

File: usaspending_api/awards/v2/filters/sub_award.py [hunk 1]
```diff
@@ -90,8 +90,9 @@ def keyword_parse(keyword):
             logger.info('Found {} transactions based on keyword: {}'.format(len(transaction_ids), keyword))
             transaction_ids = [str(transaction_id) for transaction_id in transaction_ids]
             queryset = queryset.filter(latest_transaction_id__isnull=False)
-            queryset &= queryset.extra(
-                where=['"latest_transaction_id" = ANY(\'{{{}}}\'::int[])'.format(','.join(transaction_ids))])
+
+            sql_fragment = '"subaward_view"."latest_transaction_id" = ANY(\'{{{}}}\'::int[])'
```
Comment (BrianZito) [inline usaspending_api/awards/v2/filters/sub_award.py]: I'm not that experienced with Postgres, but `'{{{}}}'::int[]` seems like code that justifies an explanatory comment.

File: usaspending_api/awards/v2/filters/sub_award.py [hunk 2]
```diff
@@ -90,8 +90,9 @@ def keyword_parse(keyword):
...
+            queryset &= queryset.extra(where=[sql_fragment.format(','.join(transaction_ids))])
```
Comment (BrianZito) [inline usaspending_api/awards/v2/filters/sub_award.py]: I see no protection against SQL injection here, and Django's documentation on .extra clearly states it DOESN'T protect against that. https://docs.djangoproject.com/en/2.1/ref/models/querysets/

Comment (tony-sappe) [reply usaspending_api/awards/v2/filters/sub_award.py]: Good eye on using .extra(). The list of `transaction_ids` are from the ES index, not user-provided. So risk of SQL injection is very low

File: usaspending_api/awards/v2/filters/sub_award.py (hunk 1 above)
Comment (tony-sappe) [reply usaspending_api/awards/v2/filters/sub_award.py]: The curly brackets are python string format syntax: https://docs.python.org/3.3/library/string.html#format-string-syntax

The inner `{}` is for the list of transaction ids, and the `{{` ..`}}` are for including a singe `{}` around the transaction ids in the final string.

I can add a comment

**Pull request merged**